ArduCopter/Log.cpp

```diff
@@ -323,6 +334,24 @@ void Copter::Log_Write_Heli()
         control_output          : motors->get_control_output(),
     };
     logger.WriteBlock(&pkt_heli, sizeof(pkt_heli));
+
+    // Write a log message per swashplate instance
+    for (uint8_t i=0; i<motors->MAX_SWASHPLATES; i++) {
+        // If we have written data to that instance of swashplate we log it
+        float col = 0.0, tcyc = 0.0, pcyc = 0.0, rcyc = 0.0;
+        if (motors->get_swash_angles(i, col, tcyc, pcyc, rcyc)) {
+            struct log_Heli_Swash pkt_heli_swash = {
+                LOG_PACKET_HEADER_INIT(LOG_HELI_SWASH_MSG),
+                time_us                 : AP_HAL::micros64(),
+                instance                : i,
+                collective_angle        : col,
+                total_cyclic_angle      : tcyc,
+                pitch_cyclic_angle      : pcyc,
+                roll_cyclic_angle       : rcyc,
+            };
+            logger.WriteBlock(&pkt_heli_swash, sizeof(pkt_heli_swash));
+        }
+    }
```

Comment:
This functionality could be done within the existing `motors->Log_Write()` function which is also called at 10Hz.